src/main/java/network/brightspots/rcv/GuiConfigController.java

```diff
@@ -839,10 +841,13 @@ private String getChoiceElse(ChoiceBox choiceBox, Enum defaultValue) {
     return choiceBox.getValue() != null ? choiceBox.getValue().toString() : defaultValue.toString();
   }
 
+  // creates a new rawContestConfig object from the contents of the UI widgets
+  // used when creating a new config and to see if we need to save the currently loaded config
   private RawContestConfig createRawContestConfig() {
     RawContestConfig config = new RawContestConfig();
 
     OutputSettings outputSettings = new OutputSettings();
+    // textFields default to returning "" when no text has been set which is expected
```

Comment:
Unnecessary comment.

Reply:
>    I believe this is because of the isCdf() function recently added to RawContestConfig; I think we should move that to ContestConfig instead

Yeah, that sounds right.  

>    code should be "" instead of null (may require an additional check in the main tabulation logic to know to ignore "" strings for codes)
>    provider should also be "" instead of null
>    Then you should also be able to replace any instances of != null for these two throughout the code to call the get directly, like you did for contestDate

I didn't get these (wtf???) but I suspect if I move the "" default logic into RawContestConfig will be fixed.

>    Fix test2013MinneapolisMayorScale()
Nooo???

>    In future versions, we might change what "null" means for these fields. I prefer that they're made explicit. I'll bitch about it in #211.

Ok, will continue there.